flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/TaskManagerRunner.java

```diff
@@ -306,10 +308,8 @@ public static void runTaskManager(Configuration configuration, ResourceID resour
 		taskManagerRunner.start();
 	}
 
-	public static void runTaskManagerSecurely(String[] args, ResourceID resourceID) {
+	public static void runTaskManagerSecurely(Configuration configuration, ResourceID resourceID) {
 		try {
-			final Configuration configuration = loadConfiguration(args);
```

Comment:
I would put the `ResourceID` generation here instead of moving the configuration loading out of this method.